packages/net/src/provider/blocking.rs

```diff
@@ -0,0 +1,74 @@
+use crate::{NetProvider, RequestHandler, USER_AGENT};
+use std::cell::RefCell;
+use std::io::Read;
+use thiserror::Error;
+use url::Url;
+
+const FILE_SIZE_LIMIT: u64 = 1_000_000_000; // 1GB
+
+pub struct SyncProvider<T>(pub RefCell<Vec<T>>);
```

Comment:
Not sure what I think about the blocking client. I think a `ureq` option is a nice-to-have, but I feel like even simple use cases of Blitz will want to fetch network requests in parallel (which would imply using a ureq with a thread pool), and I'm not sure if it's worth the maintenance cost at this stage (we could also add a blocking wrapper around the non blocking client for cases like the screenshot example).

Is binary size optimisation the motivation for including a ureq-based client?